CMakeLists.txt

```diff
@@ -31,52 +30,50 @@ if(${CMAKE_GENERATOR} STREQUAL "Ninja")
     endif()
 endif()
 
-# Default Blizzard Community Platform API key
-if (DEFINED SC_DEFAULT_APIKEY)
-    add_compile_options(-DSC_DEFAULT_APIKEY="${SC_DEFAULT_APIKEY}")
+# Install paths
+if (SC_USE_FLAT_INSTALL)
+  set(SIMC_INSTALL_BIN ".")
+  set(SIMC_INSTALL_SHARED ".")
+else()
+  set(SIMC_INSTALL_BIN "bin")
+  set(SIMC_INSTALL_SHARED "share/SimulationCraft/SimulationCraft")
 endif()
 
-# output a compile_commands.json used for editor tooling
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+# helper function to enable warnings on targets
+function(sc_enable_warnings target)
+  target_compile_options(${target} PRIVATE
+    $<$<CXX_COMPILER_ID:MSVC>:/W4>
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -Wpedantic>
+  )
+endfunction()
+
+function(sc_common_compiler_options target)
+  sc_enable_warnings(${target})
+  
+  # Require /permissive- when building with Visual Studio
+  target_compile_options(${target} PRIVATE
+    $<$<CXX_COMPILER_ID:MSVC>:/permissive->
+  )
+endfunction()
```

Comment:
I'm not sure it's worth making a function for this considering you then call it for all targets. But even if it is there is no point in making it 2 functions.

Reply:
While there aren't any plans to ever export a simc engine library, I thought it couldn't hurt to properly set it as a target property.

You are right in that it is a bit verbose :-)